sharepoint-adfs/azuredeploy.json

```diff
@@ -0,0 +1,794 @@
+      }
+    },
+    "spPassphrase": {
+      "type": "securestring",
+      "minLength": 1,
+      "metadata": {
+        "description": "Passphrase of the SharePoint farm"
+      }
+    },
+    "vmDCSize": {
+      "type": "string",
+      "defaultValue": "Standard_F4",
+      "metadata": {
+        "description": "Size of the DC VM"
+      }
+    },
+    "vmSQLSize": {
+      "type": "string",
+      "defaultValue": "Standard_A4_v2",
+      "metadata": {
+        "description": "Size of the SQL VM"
+      }
+    },
+    "vmSPSize": {
+      "type": "string",
+      "defaultValue": "Standard_A8_v2",
+      "metadata": {
+        "description": "Size of the SQL VM"
+      }
+    },
+    "vmsTimeZone": {
+      "type": "string",
+      "defaultValue": "Central European Standard Time",
+      "minLength": 2,
+      "metadata": {
+        "description": "Time zone of the VMs. Get the list with '[System.TimeZoneInfo]::GetSystemTimeZones().Id'. Note that 'UTC' works but 'UTC+xx' does NOT work."
+      }
+    },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+      },
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/sharepoint-adfs/"
+    }
+  },
+  "variables": {
+    "generalSettings": {
+      "storageAccountName": "[concat('storage', uniqueString(resourceGroup().id))]",
+      "vmsStorageAccountContainerName": "vhds",
+      "dscScriptsFolder": "dsc",
+      "vmDCName": "DC",
+      "vmSQLName": "SQL",
+      "vmSPName": "SP"
+    },
+    "vNetSettings": {
+      "vNetPrivateName": "[concat(resourceGroup().name, '-vnet')]",
+      "vNetPrivatePrefix": "10.0.0.0/16",
+      "vNetPrivateSubnet1Name": "Subnet-1",
+      "vNetPrivateSubnet1Prefix": "10.0.1.0/24",
+      "vNetPrivateSubnet2Name": "Subnet-2",
+      "vNetPrivateSubnet2Prefix": "10.0.2.0/24",
+      "vNetPrivateSubnet3Name": "Subnet-3",
+      "vNetPrivateSubnet3Prefix": "10.0.3.0/24"
+    },
+    "vmDC": {
+      "vmImagePublisher": "MicrosoftWindowsServer",
+      "vmImageOffer": "WindowsServer",
+      "vmImageSKU": "2016-Datacenter",
+      "vmOSDiskName": "vm-DC-OSDisk",
+      "vmDataDiskName": "vm-DC-DataDisk",
+      "vmVmSize": "[parameters('vmDCSize')]",
+      "vmNicName": "[concat(variables('generalSettings').vmDCName, '-', 'nic-0')]",
+      "nicPrivateIPAddress": "10.0.1.4",
+      "vmPublicIPName": "vmDC-PublicIP",
+      "vmPublicIPDnsName": "[toLower(replace(concat(resourceGroup().name, '-', variables('generalSettings').vmDCName), '_', '-'))]"
```

Comment:
best not to use the rg name in the dns name - resourceGroup names can contain chars not allowed in a dns name.

Reply:
Indeed, I noticed I had to convert the '_' into '-' to avoid errors in travis test deployment, but I did that because:
- it's very convenient: just 1 value to type instead of 4
- I think the risk is low with the string functions I used to format the domainNameLabel 
- most people would likely use resourceGroup name in domainNameLabel as it's a good guarantee of its uniqueness
- even if domainNameLabel were split in separate params, template wouldn't validate it and deployment would anyway fail the same way as in current version.

With that in mind, please let me know if I should still change as you requested, or if I can leave it this way?
PS: I tried to find the exact format of [domainNameLabel](https://docs.microsoft.com/en-us/azure/templates/microsoft.network/publicipaddresses) but I couldn't...

Reply:
Resource group names can also contain parens and periods and while it's not common we do see failures when using this pattern.  Length is also a problem - so it's a practice we want to avoid....

I think DNS is just alpha-numeric and a hyphen.

Reply:
Understood, thank you for the explanation, I changed template as requested and added parameter dnsPrefix to set domainNameLabel separately from resource group name